src/stats/stats.ts

```diff
@@ -0,0 +1,85 @@
+import _ from "lodash";
+
+import { FrameEntryType, PlayerIndexedType, Frames, FramesType } from "./common";
+
+export interface StatComputer<T> {
+    setPlayerIndices(indices: PlayerIndexedType[]): void;
+    processFrame(newFrame: FrameEntryType, allFrames: FramesType): void;
+    fetch(): T;
+}
+
+export interface StatOptions {
+    processOnTheFly: boolean;
+}
+
+const defaultOptions: StatOptions = {
+    processOnTheFly: false,
+}
+
+export class Stats {
+    private options: StatOptions;
+    private lastProcessedFrame: number | null = null;
+    private lastFrame: number;
+    private frames: FramesType = {};
+    private opponentIndices = new Array<PlayerIndexedType>();
+    private allComputers = new Array<StatComputer<unknown>>();
+
+    public constructor(options?: StatOptions) {
```

Comment:
I don't really think this option is necessary. It's not settable by the user of the lib atm is it?

In any case, if we do want to keep it, usually I feel like what I'm used to seeing when libraries do this is that each field in the options object is optional. Basically something like:
```
export interface StatOptionsInput {
    processOnTheFly: boolean?;
}
```
And then using: `public constructor(options: StatOptionsInput = {})`
What do you think?

Reply:
Removed the options in commit ee6424.